src/schema/validator.js

```diff
@@ -94,9 +152,26 @@ const _validateAddon = validator.compile({
   $ref: '#/types/WebExtensionManifest',
 });
 
-export const validateAddon = (...args) => {
-  const isValid = _validateAddon(...args);
-  validateAddon.errors = filterErrors(_validateAddon.errors);
+const _validateAddonMV3 = validator.compile({
+  ...schemaObject,
+  id: 'manifest-v3',
+  $merge: {
+    source: { $ref: '#/types/ManifestBase' },
+    with: {
+      properties: {
+        manifest_version: { maximum: 3 },
+      },
+    },
+  },
+});
+
+export const validateAddon = (
+  manifestData,
+  { enableManifestVersion3 } = {}
+) => {
+  const _validate = enableManifestVersion3 ? _validateAddonMV3 : _validateAddon;
```

Comment:
Note to self: this isn't going to work well enough, because we can only have one schema definition with id "manifest" but using a different one makes the merged extended definitions coming from the other APIs not being applied in the manifest v3 validation.

I've started to rework this and I'll update this PR asap.

Reply:
Refactored this in 413f14f and verified that this strategy does work as expected (while verifying that the page_action manifest property mv3 deprecation). I'll add some more test coverage to explicitly cover the part that wasn't working in the previous approach (which wasn't making any test to fail yet).